### PR TITLE
[super wip still] add run_tests command start

### PR DESCRIPTION
**Pull request by d-m-u**
Start to the run_tests command addition. 

☠️ don't look yet please 

Stealing Jason's code from https://github.com/Fryguy/miq_bot/commit/dc0cc52ca6bbbfeaf5061c6681f4e6c53b9e85ab and adding the start that's at https://github.com/ManageIQ/miq_bot/pull/447/files

commit 1 is Jason's original with the gemfile changes and the start of the logic for commit 4
commit 2 is getting dropped eventually, it's just here for the sake of current convenience
3 is the barebones command addition itself
4 is the meat of 3

### CODE REVIEW

**Comment by NickLaMuro** [comment]
> ☠️ don't look yet please

@d-m-u looking (sorry not sorry)

File: spec/lib/github_service/commands/run_tests_spec.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+           :labels         => [],
+           :repository_url => "https://api.fakegithub.com/repos/#{fq_repo_name}")
+  end
+
+  subject do
+    described_class.new(issue)
+  end
+
+  describe "create_pr" do
+    it "better stuff" do
```
Comment (NickLaMuro) [inline spec/lib/github_service/commands/run_tests_spec.rb]: Judging you based on these test names...

File: spec/lib/github_service/commands/run_tests_spec.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      allow(issue).to receive(:head).and_return(double(:sha => '222222'))
+
+      expect(subject.send(:create_pr, repo_path, issuer)).to eq([["manageiq-ui-classic"],[]])
+    end
+  end
+
+
+
+  describe "extract_repo_names" do
+    it "with things" do
```
Comment (NickLaMuro) [inline spec/lib/github_service/commands/run_tests_spec.rb]: ... still judging...

File: spec/lib/github_service/commands/run_tests_spec.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        .to_return(:status => 404, :body => "", :headers => {})
+
+      stub_request(:get, "https://api.github.com/repos/ManageIQ/manageiq-ui-classic")
+        .with(headers: {'Accept' => 'application/vnd.github.v3+json', 'Accept-Encoding' => 'gzip;q=1.0,deflate;q=0.6,identity;q=0.3', 'Content-Type'=>'application/json', 'User-Agent' => 'Octokit Ruby Gem 4.8.0'})
+        .to_return(:status => 200, :body => "", :headers => {})
+
+      expect(described_class.new(issue).send(:extract_repo_names, "ManageIQ/foo, manageiq-ui-classic")).to eq([["manageiq-ui-classic"], ["ManageIQ/foo"]])
+    end
+
+    it "with other things" do
```
Comment (NickLaMuro) [inline spec/lib/github_service/commands/run_tests_spec.rb]: ಠ\_ಠ

**Comment by Fryguy** [comment]
Closing in favor of #454

**Pull request closed without merging**